server/src/main/java/io/crate/planner/operators/Collect.java

```diff
@@ -344,7 +345,7 @@ public FetchRewrite rewriteToFetch(TableStats tableStats, Collection<Symbol> use
             } else if (!SymbolVisitors.any(Symbols.IS_COLUMN, output)) {
                 newOutputs.add(output);
                 replacedOutputs.put(output, output);
-            } else if (SymbolVisitors.any(usedColumns::contains, output)) {
+            } else if (SymbolVisitors.any(output::equals, usedColumns.stream().collect(Collectors.toList()))) {
```

Comment:
What do you think of adding another overload to `any` that takes an `Iterable` instead of a List?

Currently `any` uses a `List` to be able to do an index-access loop, but overall it would also work with a regular foreach/iterator.
That would be cheaper than to copy the items into a new list.

Reply:
Added an overload. We probably can drop the version with List (index access just gets the next) but I kept it just in case.